macros/sql_gen/gen_joined_metrics_cte.sql

```diff
@@ -107,12 +107,30 @@
     joined_metrics as (
 
         select 
-            first_join_metrics.*
+            first_join_metrics.date_{{grain}}
+
+            {% if secondary_calculations | length > 0 %}
+                {% for period in relevant_periods %}
+                    ,first_join_metrics.date_{{ period }}
+                {% endfor %}
+            {% endif %}
```

Comment:
Do you need the check on `secondary_calculations | length`? If there are no secondary calcs, there will be no `relevant_periods`, and the loop will be skipped. Right? (if that's not true, then I'll start asking questions like "what happens if there's a relevant period but it gets dropped because there's no secondary calcs")